elyra/pipeline/processor.py

```diff
@@ -352,15 +366,56 @@ def _sort_operation_dependencies(operations_by_id: dict, ordered_operations: lis
                     )
             ordered_operations.append(operation)
 
+    def add_runtime_image(self, instance, execution_object: Any, **kwargs) -> None:
```

Comment:
Should these be on `RuntimePipelineProcessor` instead since they don't apply to local execution?

Could we use a forward reference to annotate the type on `instance`?  Would a name like `runtime_processor` be more clear?
```suggestion
    def add_runtime_image(self, runtime_processor: "RuntimePipelineProcessor", execution_object: Any, **kwargs) -> None:
```

Reply:
In this case the instance refers to the `ElyraProperty` (since each individual property calls `runtime_processor.add_{property_name}(self, execution_object`). But I've gone ahead and type-hinted those anyway since it's probably for the best 😄